postgresql_watcher/watcher.py

```diff
@@ -1,126 +1,153 @@
-        if not select([conn], [], [], 5) == ([], [], []):
-            logger.debug("Casbin policy update identified..")
-            conn.poll()
-            while conn.notifies:
-                notify = conn.notifies.pop(0)
-                logger.debug(f"Notify: {notify.payload}")
-                process_conn.send(notify.payload)
+POSTGRESQL_CHANNEL_NAME = "casbin_role_watcher"
 
 
 class PostgresqlWatcher(object):
+
     def __init__(
         self,
         host: str,
         user: str,
         password: str,
-        port: Optional[int] = 5432,
-        dbname: Optional[str] = "postgres",
-        channel_name: Optional[str] = POSTGRESQL_CHANNEL_NAME,
-        start_process: Optional[bool] = True,
+        port: int = 5432,
+        dbname: str = "postgres",
+        channel_name: Optional[str] = None,
+        start_listening: bool = True,
         sslmode: Optional[str] = None,
         sslrootcert: Optional[str] = None,
         sslcert: Optional[str] = None,
         sslkey: Optional[str] = None,
         logger: Optional[Logger] = None,
-    ):
+    ) -> None:
+        """
+        Initialize a PostgresqlWatcher object.
+
+        Args:
+            host (str): Hostname of the PostgreSQL server.
+            user (str): PostgreSQL username.
+            password (str): Password for the user.
+            port (int): Post of the PostgreSQL server. Defaults to 5432.
+            dbname (str): Database name. Defaults to "postgres".
+            channel_name (str): The name of the channel to listen to and to send updates to. When None a default is used.
+            start_listening (bool, optional): Flag whether to start listening to updates on the PostgreSQL channel. Defaults to True.
+            sslmode (Optional[str], optional): See `psycopg2.connect` for details. Defaults to None.
+            sslrootcert (Optional[str], optional): See `psycopg2.connect` for details. Defaults to None.
+            sslcert (Optional[str], optional): See `psycopg2.connect` for details. Defaults to None.
+            sslkey (Optional[str], optional): See `psycopg2.connect` for details. Defaults to None.
+            logger (Optional[Logger], optional): Custom logger to use. Defaults to None.
+        """
         self.update_callback = None
         self.parent_conn = None
         self.host = host
         self.port = port
         self.user = user
         self.password = password
         self.dbname = dbname
-        self.channel_name = channel_name
+        self.channel_name = (
+            channel_name if channel_name is not None else POSTGRESQL_CHANNEL_NAME
+        )
         self.sslmode = sslmode
         self.sslrootcert = sslrootcert
         self.sslcert = sslcert
         self.sslkey = sslkey
         if logger is None:
             logger = getLogger()
         self.logger = logger
-        self.subscribed_process = self.create_subscriber_process(start_process)
+        self.parent_conn: Connection | None = None
```

Comment:
Connection | None = None is not supported in python version less than 10

Reply:
Fixed.